packages/react-instantsearch-core/src/core/createInstantSearchManager.js

```diff
@@ -72,12 +72,20 @@ export default function createInstantSearchManager({
   }
 
   function updateClient(client) {
+    if (helper.getClient() === client) {
```

Comment:
Depends on the implementation of `InstantSearch` but I would rather branch inside the component rather than inside the function. It's easier to follow at the component level, it's common to have condition around such "side effects" inside the component.